python_modules/dagster/dagster/_streamline/asset_check_health.py

```diff
@@ -133,3 +137,18 @@ async def compute_for_asset_checks(
             warning_checks=warning_checks,
             all_checks=check_keys,
         )
+
+    @classmethod
+    def _blocking_batch_load(
+        cls, keys: Iterable[AssetKey], context: LoadingContext
+    ) -> Iterable[Optional["AssetCheckHealthState"]]:
+        asset_check_health_states = None
+        if context.instance.streamline_read_asset_health_supported():
+            asset_check_health_states = context.instance.get_asset_check_health_state_for_assets(
```

Comment:
would it make sense to push the flag check down in to the method? If the results are `None`-able anyway might be reasonable